2017/opmodes/DaisyBeaconAutonomous.java

```diff
@@ -0,0 +1,211 @@
+                ptt.addData("ALLIANCE", "Blue");
+            } else if (event.kind == GamepadTask.EventKind.BUTTON_B_DOWN) {
+                selectAlliance(Alliance.RED);
+                ptt.addData("ALLIANCE", "Red");
+            } else if (event.kind == GamepadTask.EventKind.LEFT_TRIGGER_DOWN) {
+                pathChoice = AutonomousPath.CORNER_PARK;
+                ptt.addData("AUTONOMOUS", "Corner Park");
+            } else if (event.kind == GamepadTask.EventKind.RIGHT_TRIGGER_DOWN) {
+                pathChoice = AutonomousPath.CENTER_PARK;
+                ptt.addData("AUTONOMOUS", "Center Park");
+            } else if (event.kind == GamepadTask.EventKind.LEFT_BUMPER_DOWN) {
+                actionChoice = AutonomousAction.LAUNCH_1;
+                ptt.addData("AUTONOMOUS", "Launch 1 Ball");
+            } else if (event.kind == GamepadTask.EventKind.RIGHT_BUMPER_DOWN) {
+                actionChoice = AutonomousAction.LAUNCH_2;
+                ptt.addData("AUTONOMOUS", "Launch 2 Balls");
+            } else if (event.kind == GamepadTask.EventKind.BUTTON_A_DOWN) {
+                beaconChoice = AutonomousBeacon.BEACON_1;
+                ptt.addData("BEACON", "Claim 1 Beacon");
+            } else if (event.kind == GamepadTask.EventKind.BUTTON_Y_DOWN) {
+                beaconChoice = AutonomousBeacon.BEACON_2;
+                ptt.addData("BEACON", "Claim 2 Beacons");
+            }
+        }
+
+        if (e instanceof SingleShotTimerTask.SingleShotTimerEvent) {
+            conveyor.setPower(0);
+            addTask(runToPositionTask);
+        } else if (e instanceof RunToEncoderValueTask.RunToEncoderValueEvent) {
+            RunToEncoderValueTask.RunToEncoderValueEvent event = (RunToEncoderValueTask.RunToEncoderValueEvent) e;
+            if (event.kind == RunToEncoderValueTask.EventKind.DONE) {
+                if (!launched && actionChoice == AutonomousAction.LAUNCH_2) {
+                    conveyor.setPower(0.5);
+                    addTask(stt);
+                    launched = true;
+                } else {
+                    addTask(deadReckonBeaconTask);
+                }
+            }
+        }
+    }
+
+    private void selectAlliance(Alliance color)
+    {
+        if (color == Alliance.BLUE) {
+            // Do blue setup.
+            turnMultiplier = -1;
+        } else {
+            // Do red setup.
+            turnMultiplier = 1;
+        }
+    }
+
+    private FourWheelGearedDriveDeadReckon pathSetup(AutonomousPath pathChoice)
+    {
+        FourWheelGearedDriveDeadReckon path = new FourWheelGearedDriveDeadReckon(this, TICKS_PER_INCH, TICKS_PER_DEGREE,
+                frontLeft, frontRight, rearLeft, rearRight);
+
+        if (pathChoice == AutonomousPath.CORNER_PARK) {
+            path.addSegment(DeadReckon.SegmentType.STRAIGHT,  58, STRAIGHT_SPEED);
+            path.addSegment(DeadReckon.SegmentType.TURN,     120, TURN_SPEED * turnMultiplier);
```

Comment:
If you are going to pad make sure it lines up, otherwise don't pad.  For parameter lists, I actually prefer no padding, but will leave this choice as to whether to pad or not up to you.